server/src/updaters/relationship-updaters.js

```diff
@@ -91,14 +96,64 @@ async function updateRelationships(
       dbQuery(undirectedInsertQuery),
       dbQuery(directedInsertQuery),
     ];
-
     if (directedDeleteQuery.length) {
       promises.push(dbQuery(directedDeleteQuery));
     }
 
     await Promise.all(promises);
 
     return { ...errors, ...friendRequestErrors };
+  } else if (action === relationshipActions.UNFRIEND) {
+    const updateRows = userIDs.map(userID => {
+      const ids = [userID, viewer.userID].map(Number).sort((a, b) => a - b);
+      return [...ids, undirectedStatus.KNOW_OF];
+    });
+    const insertQuery = SQL`
+      INSERT INTO relationships_undirected (user1, user2, status)
+      VALUES ${updateRows}
+      ON DUPLICATE KEY UPDATE status = VALUES(status)
+    `;
+    const deleteQuery = SQL`
+      DELETE FROM relationships_directed
+      WHERE status = ${directedStatus.PENDING_FRIEND} AND 
+        user1 = ${viewer.userID} AND user2 IN (${userIDs})
+    `;
```

Comment:
I think we'll use this endpoint for rejecting somebody's friend request too, which means we should delete friend requests in either direction

Reply:
I was wondering about that. If we will do that it would be possible to decline friend request but the requester could invite again and again. But I guess that's how it works on social apps. If a user doesn't want to receive friend requests anymore he can block the annoying requester. I've updated a query to delete in both directions then